back/src/Model/GameRoom.ts

```diff
@@ -220,12 +218,20 @@ export class GameRoom {
             // If the user is part of a group:
             //  should he leave the group?
             const distance = GameRoom.computeDistanceBetweenPositions(user.getPosition(), user.group.getPosition());
-            if (distance > this.groupRadius) {
+            if (user.following.length === 0 && distance > this.groupRadius) {
                 this.leaveGroup(user);
             }
```

Comment:
The issue with "bug 1" must be here.

We are looking if the current user that is moving should leave the group, but we are not doing the same for all users of the group. We should probably make the same computation for all players in the group here.